frontend/src/utils/server-utils.ts

```diff
@@ -229,14 +239,49 @@ const createDefaultDateItem = (date: number, validity?: Validity): DateItem => {
 };
 
 /**
- * Create new array including dates specified within the validity parameter.
- *
- * @return Array of integers which represents a given date.
+ * Generate intermediate date items using start_date and end_date field
+ * for every available file ressources
  */
-const updateLayerDatesWithValidity = (layer: ValidityLayer): DateItem[] => {
-  const { dates, validity } = layer;
+async function generateIntermediateDateItemFromDataFile(
+  layerDates: number[],
+  layerPathTemplate: string,
+) {
+  const ranges: StartEndDate[] = await Promise.all(
+    layerDates.map(async r => {
+      const path = layerPathTemplate.replace(/{.*?}/g, match => {
+        const format = match.slice(1, -1);
+        return moment(r).format(format);
+      });
+
+      const res = await fetch(path);
```

Comment:
Wouldn't it be better to use the `fetchWithTimeout` here?

Reply:
I wanted to and I also wanted 2 distincts error messages depending on the error.